vello/src/scene.rs

```diff
@@ -333,24 +344,340 @@ impl<'a> DrawGlyphs<'a> {
+        let transform = conv_skrifa_transform(transform);
+        let prior_transform = self.last_transform();
+        self.transform_stack.push(prior_transform * transform);
+    }
+
+    fn pop_transform(&mut self) {
+        self.transform_stack.pop();
+    }
+
+    fn push_clip_glyph(&mut self, glyph_id: GlyphId) {
+        let Some(outline) = self.outlines.get(glyph_id) else {
+            eprintln!("Didn't get expected outline");
+            return;
+        };
+
+        let mut path = BezPathOutline(BezPath::new());
+        let draw_settings =
+            DrawSettings::unhinted(skrifa::instance::Size::unscaled(), [].as_slice());
+
+        let Ok(_) = outline.draw(draw_settings, &mut path) else {
+            return;
+        };
+        self.clip_depth += 1;
+        self.scene
+            .push_layer(Mix::Clip, 1.0, self.last_transform().to_kurbo(), &path.0);
+    }
+
+    fn push_clip_box(&mut self, clip_box: skrifa::raw::types::BoundingBox<f32>) {
+        let clip_box = Rect::new(
+            clip_box.x_min.into(),
+            clip_box.y_min.into(),
+            clip_box.x_max.into(),
+            clip_box.y_max.into(),
+        );
+        if self.clip_depth == 0 {
+            self.clip_box = clip_box;
+        }
+        self.clip_depth += 1;
+        self.scene
+            .push_layer(Mix::Clip, 1.0, self.last_transform().to_kurbo(), &clip_box);
+    }
+
+    fn pop_clip(&mut self) {
+        self.scene.pop_layer();
+        self.clip_depth -= 1;
+        if self.clip_depth == 0 {
+            self.clip_box = DEFAULT_CLIP_RECT;
+        }
+    }
+
+    fn fill(&mut self, brush: skrifa::color::Brush<'_>) {
+        let brush = conv_brush(brush, self.cpal);
+        self.scene.fill(
+            Fill::NonZero,
+            Affine::IDENTITY,
+            &brush,
+            Some(self.last_transform().to_kurbo()),
+            &self.clip_box,
+        );
+    }
+
+    fn push_layer(&mut self, composite: skrifa::color::CompositeMode) {
+        let blend = match composite {
+            skrifa::color::CompositeMode::Clear => Compose::Clear,
+            skrifa::color::CompositeMode::Src => Compose::Copy,
+            skrifa::color::CompositeMode::Dest => Compose::Dest,
+            skrifa::color::CompositeMode::SrcOver => Compose::SrcOver,
+            skrifa::color::CompositeMode::DestOver => Compose::DestOver,
+            skrifa::color::CompositeMode::SrcIn => Compose::SrcIn,
+            skrifa::color::CompositeMode::DestIn => Compose::DestIn,
+            skrifa::color::CompositeMode::SrcOut => Compose::SrcOut,
+            skrifa::color::CompositeMode::DestOut => Compose::DestOut,
+            skrifa::color::CompositeMode::SrcAtop => Compose::SrcAtop,
+            skrifa::color::CompositeMode::DestAtop => Compose::DestAtop,
+            skrifa::color::CompositeMode::Xor => Compose::Xor,
+            skrifa::color::CompositeMode::Plus => Compose::Plus,
+            // TODO:
```

Comment:
Yes, this conversion is annoying since the COLR spec conflates color mixing and composition.